compiler/src/iree/compiler/Dialect/Flow/Transforms/CollapseDimensions.cpp

```diff
@@ -408,15 +408,22 @@ static bool collapseDimensions(IRRewriter &rewriter,
   if (!genericOp.has_value())
     return false;
 
-  // Step 2. Check whether it is possible to collapse
+  // Step 2. Don't collapse dims if there is a clonable dequantization like
+  // producer. Collapsing in this case will prevent fusion later.
+  if (llvm::any_of(getCloneableOps(regionOp), [&](Operation *op) {
```

Comment:
This is kind of OK, but this logic seems better suited to incorporated into the `findRootOp` method?

Reply:
I think we can just leave this here for now, since we plan to remove it by switching the pass order of collapseDims later anyway

Reply:
Can we do that first... Just send out a simple PR that does the change in order. If that passes, and then you can drop all this logic, then we dont need to worry about this at all.